hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/DefaultClientConnectionStrategy.java

```diff
@@ -56,25 +53,10 @@ public void start() {
 
     @Override
     public void beforeGetConnection(Address target) {
-        if (clusterConnectorService.isClusterAvailable()) {
-            return;
-        }
-        if (asyncStart && !disconnectedFromCluster) {
```

Comment:
This will still be there bu using this check:
if (asyncStart && activeConnections.size()== 0)